src/core/shader.js

```diff
@@ -96,10 +98,30 @@ Shader.prototype = {
     var schema = this.schema;
     dataKeys.forEach(processData);
     function processData (key) {
+      var varKey = key;
+      var materialKey = '_texture_' + key;
```

Comment:
the `_texture_` prefix is to avoid namespace clashes on the material properties.